src/overlay/Peer.cpp

```diff
@@ -585,10 +646,29 @@ Peer::recvMessage(StellarMessage const& stellarMsg)
     switch (stellarMsg.type())
     {
     // group messages used during handshake, process those synchronously
-    case MessageType::HELLO:
-    case MessageType::AUTH:
+    case HELLO:
+    case AUTH:
         Peer::recvRawMessage(stellarMsg);
         return;
+    case SEND_MORE:
+    {
+        cat = "CTRL";
+        // Set this once: peers aren't allowed to switch between modes
```

Comment:
it doesn't make sense to check `ENABLE_OVERLAY_FLOW_CONTROL` as we're processing a flow control request from another peer. `ENABLE_OVERLAY_FLOW_CONTROL` reflects flow control preference of the _local_ node. We can, however, check for `mRemoteOverlayVersion` (there's already a check like that in `recvSendMore`, but it could be moved higher up here)

Reply:
We need both nodes to opt-in: we don't want experimental code to be enabled on production systems.

Reply:
and even later, I don't think we want to allow connections to have flow control only in one direction: it's more things to test